consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -186,23 +186,26 @@ impl<T: Payload> RecoveryData<T> {
         // sort by round to guarantee the topological order of parent <- child
         blocks.sort_by_key(Block::round);
         let root_from_consensus = {
-            let id_to_round: HashMap<_, _> = blocks
```

Comment:
I find the original logic a bit more intuitive, but would let @zekun000 comment here.

Reply:
Re-simplified, thanks for the pushback.